### PR TITLE
Update to accommodate latest version of locust

### PR DESCRIPTION
**Pull request by jayudey-wf**
First off I'm a big fan 😄 , this is my attempt at making the library compatible w/ the most recent versions of locust

### Problem
Swarm doesn't work with the latest versions of locust, https://github.com/anhldbk/swarm/issues/26

### Solution
Update to handle some of the architectural changes that came with locust
- item one
  - previously it looks like the 'master' was passing in the total number of clients and hatch rate to the workers
  - currently it looks like that information is passed down via the `userClassesCount` and the total number of workers is incremented from there
- item two
  - for the initial ready check locust is expecting a data field where previously it wasn't (this prompted the changes to the visitor and messages class)

this was heavily influenced by reviewing what was done in locust4j

### CODE REVIEW

**Comment by anhldbk** [comment]
@jayudey-wf Thank you for this MR. Gonna have a look at it.

File: src/main/java/com/bigsonata/swarm/interop/Message.java [hunk 1]
```diff
@@ -76,9 +40,63 @@ public Message(byte[] bytes) throws IOException {
...
+          break;
+        case NIL:
+          value = null;
+          unpacker.unpackNil();
+          break;
+        case STRING:
+          value = unpacker.unpackString();
+          break;
+        case MAP:
+          value = unpackMap(unpacker);
```
Comment (jeffhall-wk) [inline src/main/java/com/bigsonata/swarm/interop/Message.java:76]: Recursion!

**Review by jeffhall-wk — APPROVED** [review]

File: src/main/java/com/bigsonata/swarm/interop/Message.java (hunk 1 above)
Comment (anhldbk) [reply src/main/java/com/bigsonata/swarm/interop/Message.java:76]: @jayudey-wf WDYT?

Comment (jayudey-wf) [reply src/main/java/com/bigsonata/swarm/interop/Message.java:76]: i 👀 how `locust4j` solved this issue and one of the places that it looked like it changed was this approach so i thought to update it in line w/ their approach

**Comment by anhldbk** [comment]
@jayudey-wf LGTM. Thank you!

**Pull request merged**